documentation/src/main/docs/config/mappings.md

```diff
@@ -389,6 +389,36 @@ maps to the `Map<String, List<Alias>> aliases()` member, where `localhost` is th
 
     They `Map` key part in the configuration property name may require quotes to delimit the key.
 
+### `@WithUnnamedKey`
+
+The `io.smallrye.config.WithUnnamedKey` annotation allows to omit a single map key in the configuration path:
+
+```java
+@ConfigMapping(prefix = "server")
+public interface Server {
+    @WithUnnamedKey("localhost")
+    Map<String, Alias> aliases();
+    
+    interface Alias {
+        String name();
+    }
+}
+```
+
+```properties
+server.aliases.name=localhost
+server.aliases.prod.name=prod
+```
+
+The `sever.aliases.name` is an unnamed `Map` property, because it does not contain the `Map` key to populate the `Map` 
+entry. Due to `@WithUnnamedKey("localhost")` the `Map` key is not required in the configuration path. The key used to 
+look up the Map entry is given by `io.smallrye.config.WithUnnamedKey#value`:
+
+```java
+Server server = config.getConfigMapping(Server.class);
+Map<String, Alias> localhost = server.aliases.get("localhost");
+```
+
```

Comment:
It should also be explained what happens if the unnamed key (in this case `localhost`) is explicitly mentioned in the map (what is the precedence order, or is there an error, etc.). Also it could be good to show what a YAML example of this would look like.

Reply:
Correct. I forgot to clarify that.

At the moment, the unnamed key always has precedence over the explicit key with the same name, meaning that properties that use an explicit key with the same name as the unnamed key will get effectively ignored. I've consider  reading both and merge them, but I thought it would be a bit confusing to mix both, and don't have clarity when looking into an `application.properties` file.

I can probably add a warning message for such cases, instead of silently ignoring them.

Reply:
I guess it's one more argument for defaulting to a `null` map key, since that's the only way we could use the feature without preventing users from using a particular key?

(Though don't take this wrong, the limitation is perfectly fine in most cases)